sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala

```diff
@@ -125,6 +125,12 @@ private[sql] object ArrowUtils {
             largeVarTypes)).asJava)
       case udt: UserDefinedType[_] =>
         toArrowField(name, udt.sqlType, nullable, timeZoneId, largeVarTypes)
+      case _: VariantType =>
```

Comment:
`VariantType` utilizes the `metadata` field to distinguish with `StructType`

also cc @cloud-fan @hvanhovell @ueshin since it touches `ArrowUtils`